pysnmp/carrier/asyncore/stream/base.py

```diff
@@ -0,0 +1,187 @@
+#
+# This file is part of pysnmp software.
+#
+# Copyright (c) 2005-2018, Ilya Etingof <etingof@gmail.com>
+# License: http://snmplabs.com/pysnmp/license.html
+#
+# Author: Edgar Sousa <edg@edgsousa.xyz>
+#
+import errno
+import socket
+import struct
+import sys
+
+from pysnmp import debug
+from pysnmp.carrier import sockmsg, error
+from ..dgram.base import DgramSocketTransport
+
+# Ignore these socket errors
+sockErrors = {errno.ESHUTDOWN: True,
+              errno.ENOTCONN: True,
+              errno.ECONNRESET: False,
+              errno.ECONNREFUSED: False,
+              errno.EAGAIN: False,
+              errno.EWOULDBLOCK: False,
+              errno.WSAEWOULDBLOCK: False,}
+
+if hasattr(errno, 'EBADFD'):
+    # bad FD may happen upon FD closure on n-1 select() event
+    sockErrors[errno.EBADFD] = True
+
+
+class StreamSocketTransport(DgramSocketTransport):
```

Comment:
This whole class seems to be largely a copy of `DgramSocketTransport`. Could we make this common code shared between stream and dgram classes? May be by pushing it to `AbstractSocketTransport`?